File: 0242-valid-anagram/0242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        if(s.length() != t.length()) return false;

        Map<Character, Integer> sMap = new HashMap<>();
        Map<Character, Integer> tMap = new HashMap<>();

        int len = s.length();

        for(int i = 0 ; i < len ; i++)
        {
            sMap.put(s.charAt(i), sMap.getOrDefault(s.charAt(i), 0) + 1);
            tMap.put(t.charAt(i), tMap.getOrDefault(t.charAt(i), 0) + 1);
        }
        return sMap.equals(tMap);
    }
}

File: 0242-valid-anagram/0242-valid-anagram.c
bool isAnagram(char* s, char* t) {
    int count[26] = {0};

    if(strlen(s) != strlen(t)) return false;

    for(int i = 0 ; i < strlen(s) ; i++){
        count[s[i] - 'a']++;
        count[t[i] - 'a']--;
    }

    for(int i = 0 ; i < 26 ; i++){
        if(count[i] != 0){
            return false;
        }
    }

    return true;
}